### PR TITLE
Handle multi-page maps

### PR DESCRIPTION
**Pull request by cmenguy**
After discussing in the issue, it sounded like the 2nd dictionary was part of the value column of my map, but it was being accessed in the 1st column.

Upon further inspection, I noticed that `num` was being reset after `_assemble_objects` and the condition for breaking out of the `while` loop was not breaking out when it should.

After making this fix, I was able to read the file. I am not 100% sure if all pages are read completely, but number of rows seems correct at 70399.

File for reference below:
[multidict-page.snappy.parquet.zip](https://github.com/dask/fastparquet/files/2337646/multidict-page.snappy.parquet.zip)

### CODE REVIEW

**Comment by cmenguy** [comment]
Sure, I'll try to do the test today, but I think there might be something else, as only the first 1289 rows are being read correctly, and everything else is null. I'll look more into it, and happy to hear if you have any suggestions.

File: fastparquet/core.py [hunk 1]
```diff
@@ -249,7 +249,7 @@ def read_col(column, schema_helper, infile, use_cat=False,
             null = not schema_helper.is_required(cmd.path_in_schema[0])
             null_val = (se.repetition_type !=
                         parquet_thrift.FieldRepetitionType.REQUIRED)
-            num = encoding._assemble_objects(assign, defi, rep, val, dic, d,
```
Comment (martindurant) [inline fastparquet/core.py]: So the output of `_assemble_objects` should indeed be the number of rows, but L294 has another `num +=` based on the number of *values*, which is why, presumably, it looks we are done prematurely.

**Comment by cmenguy** [comment]
Alright @martindurant I spent a bit more time and was able to got it working for the whole file with this fix. Also added a test that ensures first and last lines are correct, and that every line got converted.

**Comment by martindurant** [comment]
Yes, I think this is exactly right.
Thanks for all your digging!

If you happen to also come across data which really has multiple dictionary pages, then it would be good to apply my fix to that case.

**Pull request merged**